packages/flame/lib/src/game/game_render_box.dart

```diff
@@ -2,14 +2,55 @@ import 'package:flame/src/game/game_loop.dart';
 import 'package:flame/src/game/mixins/game.dart';
 import 'package:flutter/rendering.dart';
 import 'package:flutter/widgets.dart' hide WidgetBuilder;
-//ignore_for_file: unnecessary_non_null_assertion
+
+class RenderGameWidget extends LeafRenderObjectWidget {
+  final Game game;
+
+  const RenderGameWidget({
+    Key? key,
+    required this.game,
+  }) : super(key: key);
+
+  @override
+  RenderBox createRenderObject(BuildContext context) {
+    return GameRenderBox(game, context);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, GameRenderBox renderObject) {
+    renderObject
+      ..game = game
+      ..buildContext = context;
+  }
+}
 
 class GameRenderBox extends RenderBox with WidgetsBindingObserver {
-  BuildContext buildContext;
-  Game game;
+  GameRenderBox(this._game, this.buildContext);
+
   GameLoop? gameLoop;
 
-  GameRenderBox(this.buildContext, this.game);
+  BuildContext buildContext;
+
+  Game _game;
+
+  Game get game => _game;
+
+  set game(Game value) {
+    // Identities are equal, no need to update.
+    if (_game == value) {
+      return;
+    }
+
+    if (attached) {
+      detachGame();
+    }
+
+    _game = value;
+
+    if (attached) {
+      attachGame(owner!);
+    }
```

Comment:
Nit:
```suggestion
    if (attached) detachGame();
    _game = value;
    if (attached) attachGame(owner!);
```

I can't remember if flame lint had a rule that prevented avoiding {}

Reply:
It does